nixos/modules/config/users-groups.nix

```diff
@@ -568,5 +579,17 @@ in {
   imports =
     [ (mkAliasOptionModule [ "users" "extraUsers" ] [ "users" "users" ])
       (mkAliasOptionModule [ "users" "extraGroups" ] [ "users" "groups" ])
+      {
+        environment = {
+          etc = mapAttrs' (name: { packages, ... }: {
+            name = "per-user-pkgs/${name}";
+            value.source = pkgs.symlinkJoin {
+              name = "per-user-pkgs.${name}";
+              paths = packages;
+            };
+          }) cfg.users;
```

Comment:
This will create a symlink tree + /etc/per-user-pkgs symlink for every user, even if `packages` is empty, right? Might be nice to filter out users with empty `packages`. And maybe also filter out system users.